huion-tablet-driver.py

```diff
@@ -259,6 +266,75 @@ def multi_monitor():
         main.settings['screen_width'], main.settings['screen_height'],
         main.settings['tablet_offset_x'], main.settings['tablet_offset_y']))
 
+
+# -----------------------------------------------------------------------------
+def multi_pointer(xinput_device_keyword=XINPUT_DEVICE_KEYWORD):
+    """
+    """
+
+    if not main.settings['enable_multi_pointer']:
+        return
+
+    print("\nSetting up multiple pointers. . . ")
+
+    # Removing previous existing master
+    cmd = "xinput --remove-master HuionTablet"
+    if main.settings['debug_mode']:
+        print('» {}'.format(cmd))
+    os.popen(cmd)
+
+    if main.settings['enable_multi_pointer']:
+        print("Create master input. . .")
+        cmd = "xinput create-master 'HuionTablet'"
+        if main.settings['debug_mode']:
+            print('» {}'.format(cmd))
+        os.popen(cmd)
+
+        xinput_device = os.popen("xinput | grep '%s' | cut -f 2" % xinput_device_keyword).read()
+        xinput_device = xinput_device.split('\n')[0][3:]    # Chose the first instance of keyword, strips off "id="
+        master = os.popen("xinput | grep 'HuionTablet pointer' | cut -f 2").read()
+        master = master.split('\n')[0][3:]
+        if main.settings['debug_mode']:
+            print('» input device = {}'.format(xinput_device))
+   
+        cmd = "xinput reattach {} {}".format(xinput_device, master)
+        if main.settings['debug_mode']:
+            print('» {}'.format(cmd))
+        os.popen(cmd)
+
+
+# -----------------------------------------------------------------------------
+def calibrate_mapping(xinput_device_keyword=XINPUT_DEVICE_KEYWORD):
+
+    try:
+        # Locate tablet pen device number
+        xinput_device = os.popen("xinput | grep '%s' | cut -f 2" % xinput_device_keyword).read()
+        xinput_device = xinput_device.split('\n')[0][3:]    # Chose the first instance of keyword, strips off "id="
+        
+        # Locate output device name
+        xoutput_devices = os.popen("xrandr --listactivemonitors | sort | cut -f 6 -d ' ' ").read()
+        xoutput_devices = list(filter(None, xoutput_devices.split('\n'))) # Assumes the first device is the main monitor and the second is the graphic monitor
+        xoutput_device = xoutput_devices[1]
```

Comment:
Assuming that there is always 2 monitors is unsafe.
It would be safer to get the last one in the list, but we'd still assume that the last monitor is the graphic monitor.
```suggestion
        xoutput_device = xoutput_devices[-1]
```
The best solution would be to match the monitor with the device, but I'm exactly sure on how to do that...

Reply:
I've integrated a merge request from @shervinsahba that does this, I think.